svsim/src/main/scala/Workspace.scala

```diff
@@ -0,0 +1,389 @@
+      )
+      l()
+      l(".PHONY: clean simulation replay")
+      l()
+      //format: off
+      // For this debug flow, we rebuild the simulation from scratch every time, to avoid issues if the simulation was originally compiled in a different environment, like using SiFive's `wake`.
+      l("clean:")
+	    l("\tls . | grep -v Makefile | grep -v execution-script.txt | xargs rm -rf")
+      l()
+      l("simulation: clean")
+      l("\t$(compilerEnvironment) \\")
+	    l("\t", invocationSettings.compilerPath, " \\")
+      for (argument <- invocationSettings.compilerArguments) {
+        val sanitizedArugment = argument
+          .replace("$", "$$")
+          .replace("'", "'\\''")
+          .replace(workingDirectoryPath, "$(shell pwd)")
+      l("\t\t'", sanitizedArugment, "' \\")
+      }
+      l("\t\t$(sourcefiles)")
+      l()
+      l("replay: simulation")
+      val executionScriptPath = "$(shell pwd)/execution-script.txt"
+      customSimulationWorkingDirectory match {
+        case None =>
+        case Some(value) => {
+          // Calculate relative path, to avoid being broken by wake's directory shenanigans
+          val relativePath = workingDirectory.toPath().relativize(Paths.get(value)).toString()
+          l("\tcd ", relativePath, " && \\")
+        }
+      }
+      l("\tcat ", executionScriptPath, " | { grep '^#' || true; } && \\")
+      l("\tcat ", executionScriptPath, " | sed -n 's/^[0-9]*> \\(.*\\)/\\1/p' | \\")
+      l("\t\t$(simulationEnvironment) $(shell pwd)/simulation \\")
+      for (argument <- invocationSettings.simulationArguments) {
+      l("\t\t\t'", argument.replace("$", "$$"), "' \\")
+      }
+      l()
+      l("sourcefiles = \\")
+      for ((sourceFile, index) <- sourceFiles.zipWithIndex) {
+      l("\t'", sourceFile, "'", if (index != sourceFiles.length - 1) " \\" else "")
+      }
+      l()
+      l("compilerEnvironment = \\")
+      for (((name, value), index) <- invocationSettings.compilerEnvironment.zipWithIndex) {
+      l("\t", name, "=", value, if (index != invocationSettings.compilerEnvironment.length - 1) " \\" else "")
+      }
+      l()
+      l("simulationEnvironment = \\")
+      for (((name, value), index) <- simulationEnvironment.zipWithIndex) {
+        val sanitizedValue = value.replace(workingDirectoryPath, "$(shell pwd)")
+      l("\t", name, "=", sanitizedValue, if (index != simulationEnvironment.length - 1) " \\" else "")
+      }
+      l()
+      //format: on
+    } finally {
+      makefileWriter.close()
+    }
+
+    /**
+      * Use the generated Makefile to compile the simulation, since this exercises the Makefile codepath and makes it less likely that we will break `make replay`.
+      */
```

Comment:
This is kind of tricky... We can find a better solution to decouple the build system to leverage `mill` or `wake`.